src/objective/regression_objective.hpp

```diff
@@ -161,6 +161,55 @@ class RegressionL1loss: public RegressionL2loss {
   double eta_;
 };
 
+/*!
+* \brief MAPE regression loss
+*/
+class RegressionMAPEloss: public RegressionL2loss {
+public:
+  explicit RegressionMAPEloss(const ObjectiveConfig& config): RegressionL2loss(config) { }
+
+  explicit RegressionMAPEloss(const std::vector<std::string>& strs): RegressionL2loss(strs) {
+
+  }
+
+  ~RegressionMAPEloss() {}
+
+  void GetGradients(const double* score, score_t* gradients,
+                    score_t* hessians) const override {
+    if (weights_ == nullptr) {
+      #pragma omp parallel for schedule(static)
+      for (data_size_t i = 0; i < num_data_; ++i) {
+        const double diff = score[i] - label_[i];
+        if (diff >= 0.0f) {
+          gradients[i] = 1.0f;
+        } else {
+          gradients[i] = -1.0f;
+        }
+        hessians[i] = static_cast<score_t>(0.0);
```

Comment:
cannot use zero hessians, since the output is grad/hess.

Reply:
What should be done in this case?

Reply:
you can use some approximate loss function. e.g. 1. https://en.wikipedia.org/wiki/Symmetric_mean_absolute_percentage_error . 
2. use L2 form: loss = (y-label)^2/label .
BTW, your gradient and hess actually is the L1, not the MAPE.

Reply:
> BTW, your gradient and hess actually is the L1, not the MAPE.
Yes, I just realized that I messed up. Will update shortly